### PR TITLE
Prompt Block and UnBlock

### PR DESCRIPTION
**Pull request by ConorGriffin37**
Prompt block and unblock.

### CODE REVIEW

File: CoreScriptsRoot/CoreScripts/BlockPlayerPrompt.lua [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+					WindowTitle = "Successfully Unblocked",
+					MainText = string.format("%s has been unblocked.", playerToUnblock.Name),
+					AdditonalText = nil,
+					ConfirmationText = "Okay",
+					Image = "https://www.roblox.com/Thumbs/Avatar.ashx?x=200&y=200&userId=" ..playerToUnblock.UserId,
+				}
+			else
+				return {
+					WindowTitle = "Error Unblocking Player",
+					MainText = string.format("An error occured while unblocking %s.", playerToUnblock.Name),
```
Comment (ConorGriffin37) [inline CoreScriptsRoot/CoreScripts/BlockPlayerPrompt.lua]: occured is misspelled.

File: CoreScriptsRoot/Modules/PlayerDropDown.lua [hunk 2]
```diff
@@ -190,8 +192,33 @@ end
 
 spawn(function()
 	BlockedList = GetBlockedPlayersAsync()
+	GetBlockedPlayersCompleted = true
 end)
 
+local function getBlockedUserIdsFromBlockedList()
+	local userIdList = {}
+	for userId, _ in pairs(BlockedList) do
+		table.insert(userIdList, userId)
```
Comment (BinaryResolved) [inline CoreScriptsRoot/Modules/PlayerDropDown.lua]: Could be optimized by changing to
`local userIdList = {}`
`local insert = table.insert`
`for userId, _ in pairs(BlockedList) do`
`insert(userIdList, userId)`

**Comment by hockeychaos** [comment]
is this still going to be a thing?

**Comment by ConorGriffin37** [comment]
@hockeychaos Soon TM. Once I finish implementing the new generic modal design.

File: CoreScriptsRoot/LoadingScript.lua [hunk 3]
```diff
@@ -25,7 +25,8 @@ local function getViewportSize()
 		game.Workspace.Changed:wait()
 	end
 
-	while game.Workspace.CurrentCamera.ViewportSize == Vector2.new(0,0) do
+	while game.Workspace.CurrentCamera.ViewportSize == Vector2.new(0,0) or
```
Comment (spotco) [inline CoreScriptsRoot/LoadingScript.lua]: Leave a comment pointing to the line in cpp/lua where the initial viewport is set to a size.

**Pull request merged**